src/edu/harvard/lib/librarycloud/items/ItemDAO.java

```diff
@@ -0,0 +1,487 @@
+		query.setQuery(queryStr);
+		QueryResponse response = null;
+		try {
+			response = server.query(query);
+		} catch (SolrServerException se) {
+			log.error(se.getMessage());
+			throw new BadParameterException(se.getMessage());
+		} catch (RemoteSolrException rse) {
+			if (rse.getMessage().contains("SyntaxError")) {
+				log.error("solr syntax error");
+				throw new BadParameterException("Incorrect query syntax");
+			} else {
+				String msg = rse.getMessage().replace("_keyword", "");
+				log.error(msg);
+				throw new BadParameterException("Incorrect query syntax:" + msg);
+			}
+		}
+		List<FacetField> facets = response.getFacetFields();
+		facet = null;
+		if (facets != null) {
+			facet = new Facet();
+			List<FacetType> facetTypes = new ArrayList<FacetType>();
+			for (FacetField facetField : facets) {
+				List<FacetTerm> facetTerms = new ArrayList<FacetTerm>();
+				FacetType facetType = new FacetType();
+				facetType.setFacetName(facetField.getName());
+				List<FacetField.Count> facetEntries = facetField.getValues();
+				for (FacetField.Count fcount : facetEntries) {
+					if (fcount.getCount() > 0) {
+						FacetTerm facetTerm = new FacetTerm();
+						facetTerm.setTermName(fcount.getName());
+						facetTerm.setTermCount(fcount.getCount());
+						// System.out.println(fcount.getName() + ": " +
+						// fcount.getCount());
+						facetTerms.add(facetTerm);
+					}
+				}
+				facetType.setFacetTerms(facetTerms);
+				facetTypes.add(facetType);
+			}
+			facet.setFacetTypes(facetTypes);
+		}
+		docs = response.getResults();
+		return docs;
+	}
+
+	/**
+	 * Returns search results for a given SolrDocumentList, in mods format.
+	 * 
+	 * @param doc solr document list to build results
+	 * @return the SearchResultsMods object for this solr result
+	 * @see SearchResultsMods
+	 */
+	private SearchResultsMods buildModsResults(SolrDocumentList docs) {
+		SearchResultsMods results = new SearchResultsMods();
+		Pagination pagination = new Pagination();
+		pagination.setNumFound(docs.getNumFound());
+		pagination.setStart(docs.getStart());
+		pagination.setRows(limit);
+		ModsGroup modsGroup = new ModsGroup();
+		List<ModsType> modsList = new ArrayList<ModsType>();
+		for (final SolrDocument doc : docs) {
+			Item item = new Item();
+			ModsType modsType = null;
+			try {
+				modsType = (new ItemDAO()).getModsType(doc);
+			} catch (JAXBException je) {
+				log.error(je.getMessage());
+				je.printStackTrace();
+			}
+			modsList.add(modsType);
+		}
+		modsGroup.setItems(modsList);
+		results.setItemGroup(modsGroup);
+		results.setPagination(pagination);
+		if (facet != null)
+			results.setFacet(facet);
+		return results;
+	}
+
+	
+	/**
+	 * Returns search results for a given SolrDocumentList, in dublin core
+	 * 
+	 * @param doc solr document list to build results
+	 * @return the SearchResultsDC object for this solr result
+	 * @see SearchResultsSlim
+	 */
+	private SearchResultsDC buildDCResults(SolrDocumentList docs) {

```

Comment:
Did some refactoring to slim down these methods, may be a TO DO to further refactor later